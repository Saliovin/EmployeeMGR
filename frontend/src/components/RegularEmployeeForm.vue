<script setup lang="ts">
import type { Employee } from "@/interfaces/employee";

interface Props {
    employee: Employee;
}

const props = defineProps<Props>();
const emit = defineEmits(["submit"]);
</script>

<template>
    <form @submit.prevent="(e: Event) => emit('submit', e)">
        <div class="input-group">
            <label>First Name</label>
            <input type="text" name="first_name" v-model="props.employee.first_name" />
        </div>
        <div class="input-group">
            <label>Last Name</label>
            <input type="text" name="last_name" v-model="props.employee.last_name" />
        </div>
        <div class="input-group">
            <label>Email</label>
            <input type="email" name="email" v-model="props.employee.email" />
        </div>
        <div class="input-group">
            <label>Employee Type</label>
            <select v-model="props.employee.employee_type">
                <option selected>regular</option>
                <option>contractual</option>
            </select>
        </div>
        <div class="input-group">
            <label>Number of Leaves</label>
            <input type="number" name="number_of_leaves" v-model="props.employee.properties.number_of_leaves" />
        </div>
        <div class="input-group">
            <label>Benefits</label>
            <input type="text" name="benefits" v-model="props.employee.properties.benefits" />
        </div>
        <button type="submit">Submit</button>
    </form>
</template>
