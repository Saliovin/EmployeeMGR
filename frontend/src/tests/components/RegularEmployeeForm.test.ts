import RegularEmployeeForm from '@/components/RegularEmployeeForm.vue'
import { mount } from '@vue/test-utils'
import { describe, expect, it } from 'vitest'

describe('regular employee form', () => {
    it('base', () => {
        const form = mount(RegularEmployeeForm, {
            props: {
                employee: {
                    id: 0,
                    first_name: "test",
                    last_name: "testing",
                    email: "test@test.com",
                    employee_type: "regular",
                    properties: {},
                  }
            }
        })
        expect(form.element).toMatchSnapshot()
    })
})