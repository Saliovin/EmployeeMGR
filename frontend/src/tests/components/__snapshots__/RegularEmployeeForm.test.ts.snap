// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`regular employee form > base 1`] = `
<form>
  <div
    class="input-group"
  >
    <label>
      First Name
    </label>
    <input
      name="first_name"
      type="text"
    />
  </div>
  <div
    class="input-group"
  >
    <label>
      Last Name
    </label>
    <input
      name="last_name"
      type="text"
    />
  </div>
  <div
    class="input-group"
  >
    <label>
      Email
    </label>
    <input
      name="email"
      type="email"
    />
  </div>
  <div
    class="input-group"
  >
    <label>
      Employee Type
    </label>
    <select>
      <option
        selected=""
      >
        regular
      </option>
      <option>
        contractual
      </option>
    </select>
  </div>
  <div
    class="input-group"
  >
    <label>
      Number of Leaves
    </label>
    <input
      name="number_of_leaves"
      type="number"
    />
  </div>
  <div
    class="input-group"
  >
    <label>
      Benefits
    </label>
    <input
      name="benefits"
      type="text"
    />
  </div>
  <button
    type="submit"
  >
    Submit
  </button>
</form>
`;
